docs/sources/security.md

```diff
@@ -0,0 +1,167 @@
+| `CAP_SYS_PTRACE`         | access to `/proc/pid/exe` and friends                                                                                      |
+| `CAP_SYS_RESOURCE`       | _(kernels **< 5.11** only)_ increase the amount of locked memory available.                                                |
+| `CAP_SYS_ADMIN`          | Library-level Go trace-context propagation via `bpf_probe_write_user()` and access to BTF data by the BPF metrics exporter |
+
+## Examples
+
+The final set of required capabilities depends on the actual Beyla configuration and the type of tracers being used, as described in the section  __Beyla operation modes__ above. Here are a few examples of how to run Beyla as a non-root user.
+
+### Network Metrics (using a socket filter)
+
+#### Required capabilities:
+* `CAP_BPF`
+* `CAP_NET_RAW`
+
+```
+sudo setcap cap_bpf,cap_net_raw+ep ./bin/beyla
+BEYLA_NETWORK_METRICS=1 BEYLA_NETWORK_PRINT_FLOWS=1 bin/beyla
+```
+
+### Network Metrics (using traffic control)
+
+#### Required capabilities:
+* `CAP_BPF`
+* `CAP_NET_ADMIN`
+* `CAP_PERFMON`
+
+```
+sudo setcap cap_bpf,cap_net_admin,cap_perfmon+ep ./bin/beyla
+BEYLA_NETWORK_METRICS=1 BEYLA_NETWORK_PRINT_FLOWS=1 BEYLA_NETWORK_SOURCE=tc bin/beyla
+```
+
+### Application observability
+
+#### Required capabilities:
+* `CAP_BPF`
+* `CAP_DAC_READ_SEARCH`
+* `CAP_CHECKPOINT_RESTORE`
+* `CAP_PERFMON`
+* `CAP_NET_RAW`
+* `CAP_SYS_PTRACE`
+
+```
+sudo setcap cap_bpf,cap_dac_read_search,cap_perfmon,cap_net_raw,cap_sys_ptrace+ep ./bin/beyla
+BEYLA_OPEN_PORT=8080 BEYLA_TRACE_PRINTER=text bin/beyla 
+```
+
+### Application observability with trace context propagation
+
+#### Required capabilities:
+* `CAP_BPF`
+* `CAP_DAC_READ_SEARCH`
+* `CAP_CHECKPOINT_RESTORE`
+* `CAP_PERFMON`
+* `CAP_NET_RAW`
+* `CAP_SYS_PTRACE`
+* `CAP_NET_ADMIN`
+
+```
+sudo setcap cap_bpf,cap_dac_read_search,cap_perfmon,cap_net_raw,cap_sys_ptrace,cap_net_admin+ep ./bin/beyla
+BEYLA_BPF_ENABLE_CONTEXT_PROPAGATION=1 BEYLA_OPEN_PORT=8080 BEYLA_TRACE_PRINTER=text bin/beyla 
+```
+
+## Internal eBPF tracer capability requirement reference
+
+Below is a list of internal eBPF tracers used by Beyla and their required capabilities
+### Socket flow fetcher
+- `CAP_BPF` -> for `BPF_PROG_TYPE_SOCK_FILTER`
+- `CAP_NET_RAW` -> for creating `AF_PACKET` socket
+
+### Flow fetcher (tc)
+- `CAP_BPF` 
+- `CAP_NET_ADMIN` -> for `PROG_TYPE_SCHED_CLS`
+- `CAP_PERFMON` -> direct access to `struct __sk_buff::data` and pointer arithmetic
+
+### Watcher
+- `CAP_BPF`
+- `CAP_CHECKPOINT_RESTORE`
+- `CAP_DAC_READ_SEARCH` -> access to `/proc/self/mem` to determine kernel version
+- `CAP_PERFMON` -> for `BPF_PROG_TYPE_KPROBE`
+
+### Generic tracer
+- `CAP_BPF`
+- `CAP_DAC_READ_SEARCH`
+- `CAP_CHECKPOINT_RESTORE`
+- `CAP_PERFMON`
+- `CAP_NET_RAW` -> for creating `AF_PACKET` socket used by `beyla_socket__http_filter`
+- `CAP_SYS_PTRACE` -> access to `/proc/pid/exe` and other nodes in `/proc`
+
+### TC tracers
```

Comment:
TC tracers needs to be renamed to mean something for the end users, e.g. network monitoring in TC mode and context prop.